stdlib/dateBoundaries/dateBoundaries.flux

```diff
@@ -0,0 +1,417 @@
+//
+// ### Find the Sunday of that week
+//
+// ```no_run
+// import "dateBoundaries"
+// option location = timezone.fixed(offset: -8h)
+// option now = () => 2021-12-30T00:40:44Z
+// dateBoundaries.sunday()
+//
+// Returns {start: 2021-12-26T08:00:00Z, stop:2021-12-27T08:00:00Z }
+//
+//
+//
+//
+// ### Filter by the records taken in a day
+// ```no_run
+// import "dateBoundaries"
+// ret = sunday()
+// from(bucket: "example-bucket")
+//     |> range(start: ret.start, stop: ret.stop)
+// ```
+//
+// This will return all records from Sunday this week
+//
+// ## Metadata
+// tags: date/time
+//
+sunday = () => {
+    return _day_finder(td: date.Sunday, func: _day_formatter)
+}
+
+// month will return the timestamps of the start and end of the month in a record
+// ## Parameters
+//
+//
+// ## Examples
+//
+// ### Find the timestamps for the start and stopping point of that month
+//
+// ```no_run
+// import "dateBoundaries"
+// option now = () => 2022-05-10T10:10:00Z
+// dateBoundaries.month()
+// // Returns {start:2022-05-01T00:00:00.000000000Z, stop:2022-06-01T00:00:00.000000000Z}
+// ```
+// ### Query data from this month
+// ```no_run
+// import "dateBoundaries"
+// ret = month()
+// from(bucket: "example-bucket")
+//     |> range(start: ret.start, stop: ret.stop)
+// ```
+//
+// This will return all records from this month
+//
+// ## Metadata
+// tags: date/time
+//
+month = () => {
+    s = date.truncate(t: today(), unit: 1mo)
+    start = _timezone_convert(s: s)
+
+    return {start: start, stop: date.add(d: 1mo, to: start)}
+}
+
+// week will return the timestamps of the week start of a given timestamp truncated to the day and the ending point of that week
+// ## Parameters
+//
+// - start_sunday: Boolean to represent if the month starts on a Sunday or Monday (defaults on Monday)
+//
+// ## Examples
+//
+// ### Find the timestamps for the start and end of that week with the week starting on Monday
+//
+// ```no_run
+//
+// import "dateBoundaries"
+// option now = () => 2022-05-10T00:00:00.000010000Z
+// d = dateBoundaries.week()
+//
+// // Returns {start: 2022-05-09T00:00:00.000000000Z, stop: 2022-05-16T00:00:00.000000000Z}
+// ```
+//
+// ### Find the timestamps for the start and end of that week with the week starting on Sunday
+//
+// ```no_run
+//
+// import "dateBoundaries"
+//  option now = () => 2022-05-10T10:10:00Z
+// d = dateBoundaries.week(start_sunday:true)
+//
+// // Returns {start: 2022-05-08T00:00:00.000000000Z, stop: 2022-05-14T00:00:00.000000000Z}
+// ```
+//
+// ### Example usage
+//// ```no_run
+// ret = week()
+// from(bucket: "example-bucket")
+//     |> range(start: ret.start, stop: ret.stop)
+//
+// This will return all records from this week
+//
+//
+// ## Metadata
+// tags: date/time
+//
+week = (start_sunday=false) => {
+    return
+        if start_sunday then
+            _day_finder(td: date.Sunday, func: _week_formatter)
+        else
+            _day_finder(td: date.Monday, func: _week_formatter)
+}
```

Comment:
Similar to adding a `monthOffset` parameter to `month()`, I think we should add a `weekOffset` parameter to week. This would allow for things like:

```js
// Return start and stop timestamps of last week
dateBoundaries.week(weekOffset: -1)

// Return start and stop timestamps of 3 weeks ago
dateBoundaries.week(weekOffset: -3)

// Return start and stop timestamps of next week
dateBoundaries.week(weekOffset: 1)
```